pipeline-braille-scripts/dtbook-to-pef/src/main/resources/xml/xproc/dtbook-to-pef.convert.xpl

```diff
@@ -11,109 +11,117 @@
-    <p:option name="include-production-notes" required="true"/>
-    <p:option name="show-braille-page-numbers" required="true"/>
-    <p:option name="show-print-page-numbers" required="true"/>
-    <p:option name="force-braille-page-break" required="true"/>
-    <p:option name="toc-depth" required="true"/>
-    <!-- <p:option name="ignore-document-title" required="true"/> -->
-    <!-- <p:option name="include-symbols-list" required="true"/> -->
-    <!-- <p:option name="choice-of-colophon" required="true"/> -->
-    <!-- <p:option name="footnotes-placement" required="true"/> -->
-    <!-- <p:option name="colophon-metadata-placement" required="true"/> -->
-    <!-- <p:option name="rear-cover-placement" required="true"/> -->
-    <!-- <p:option name="number-of-pages" required="true"/> -->
-    <!-- <p:option name="maximum-number-of-pages" required="true"/> -->
-    <!-- <p:option name="minimum-number-of-pages" required="true"/> -->
-    <!-- <p:option name="sbsform-macros" required="true"/> -->
+    <p:input kind="parameter" port="parameters" sequence="true">
+        <p:inline>
+            <c:param-set/>
+        </p:inline>
+    </p:input>
 
     <!-- Empty temporary directory dedicated to this conversion -->
     <p:option name="temp-dir" required="true"/>
 
+    <p:import href="http://www.daisy.org/pipeline/modules/common-utils/library.xpl"/>
     <p:import href="http://www.daisy.org/pipeline/modules/braille/common-utils/library.xpl"/>
     <p:import href="http://www.daisy.org/pipeline/modules/braille/css-utils/library.xpl"/>
     <p:import href="http://www.daisy.org/pipeline/modules/braille/pef-utils/library.xpl"/>
     <p:import href="http://www.daisy.org/pipeline/modules/dtbook-utils/library.xpl"/>
     <p:import href="http://www.daisy.org/pipeline/modules/file-utils/library.xpl"/>
     <p:import href="http://www.daisy.org/pipeline/modules/fileset-utils/library.xpl"/>
-    <p:import href="generate-toc.xpl"/>
     
     <p:variable name="lang" select="(/*/@xml:lang,'und')[1]"/>
     
+    <!-- Ensure that there's exactly one c:param-set -->
+    <p:identity>
+        <p:input port="source">
+            <p:pipe step="main" port="parameters"/>
+        </p:input>
+    </p:identity>
+    <p:wrap-sequence wrapper="c:param-set"/>
+    <p:unwrap match="/c:param-set/c:param-set"/>
+    <p:delete match="/c:param-set/c:param[@name = following-sibling::c:param/@name]"/>
+    <p:identity name="parameters"/>
+    <p:sink/>
+    
     <px:dtbook-load name="load">
         <p:input port="source">
             <p:pipe step="main" port="source"/>
         </p:input>
     </px:dtbook-load>
+    <p:sink/>
     
-    <pxi:generate-toc>
+    <p:identity>
         <p:input port="source">
             <p:pipe step="load" port="in-memory.out"/>
         </p:input>
-        <p:with-option name="depth" select="$toc-depth"/>
-    </pxi:generate-toc>
+    </p:identity>
+    <px:message message="Generating table of contents"/>
+    <p:xslt>
```

Comment:
Why not in XProc step anymore? I know the step is very minimal but I always like to wrap p:xslt's in a step because it makes things more readable.